### DISCUSSION TITLE
Form returns 500 Server Error

### DISCUSSION THREAD

**Discussion by hanneshoepfnerio** [studio1902/statamic-peak#305] [Q&A] [answered]
Hello,
I'm having an issue with the form component. It returns a Server 500 error.
When I check the response I do get a little more detail:

Expected response code \"250/251/252\" but got code \"553\", with message \"553 5.7.1 <info@site.com>: Sender address rejected: not owned by user m067bb72\".",
    "exception": "Symfony\\Component\\Mailer\\Exception\\TransportException",
    ....
    
 The mail settings in my .env are all set and it is working. Responses are stored and I can also send the test email.
 The Error message is indicating that the from address is somehow not "belonging" to the user.
 When I change the from address to default (the hello@example) I also get an error when sending out the test mail. Also an 553.
 So therefore I'm assuming that somehow everything works, but for some reason the ajax / fetch of the form has an issue.
 
 I've tried different mail services and configurations, always same issue.
 
 Would be great to get some assistance on this topic.
 
 Thank you

**Comment by jeroenimpres** [comment]
@hanneshoepfnerio this is not a PEAK issue, but a general mail misconfiguration that could actually happen in any site. The errocode you are receiving is a mail-error code, not a HTTP one. Please make sure that you enter the correct credentials and the corresponding e-mailaddress. In this case it pretty clearly tells you that info@site.com is not owned by user m067bb72 and I also doubt hello@example.com is
Please make sure you are using the correct configuration. For further questions, it might be best to turn to stackoverflow or a site like that, as this is really not PEAK specific

**Reply by hanneshoepfnerio** [reply]
Hi @jeroenimpres,
thank you for your reply.
I will investigate further. Even though, as I said, the config is working. Mails are sending, the test is fine.
I'm not having the issue when not using the form partial. But I'll keep you posted.
Thanks!

**Comment by hanneshoepfnerio** [comment, marked as answer]
Hi everyone,
I found the issue, which might be something other users experience as well.
If you install statamic with a default peak starterkit, you have a form out of the box.
This form has a default from address set, which is NOT overwritten by the .env config.
Its only overwritten when deleted.
<img width="314" alt="Bildschirm­foto 2023-03-09 um 11 35 57" src="https://user-images.githubusercontent.com/77494707/223998823-0eccf434-5e9e-412d-8a15-d91e9d32600b.png">

Therefore the test was using the .env config, which was working, but the form was using the default peak settings which are obviously not owned by the .env config and failed.

I guess leaving them empty on default would be great.

**Reply by robdekort** [reply]
Good point!!! I'll look into it. Thanks for reporting.

**Reply by robdekort** [reply]
Unfortunately I can't use `{{ config:mail:from:address }}` in the form config. So I don't think I can improve this.

**Reply by robdekort** [reply]
It logs an error that we need a `to` header. So the Statamic code doesn't pick this up.

**Reply by hanneshoepfnerio** [reply]
Hey,
can't you just leave it unfilled?
Or as a slight improvement – I got lost because info@site.com was so generic, that I actually did not understand in the server response that this was even a real value.
Maybe changing it to "you-need-to-change-this@form-config" would help when people debugging it.

**Reply by robdekort** [reply]
Unfilled would cause a submission error which isn't a nice default either. Your suggestion is something to consider. Although I'd rather have a proper solution like using the .env property. This would need a change the Statamic Core though.
